src/interfaces/dnmdowner.hpp

```diff
@@ -0,0 +1,98 @@
+#ifndef _SRC_INTERFACES_DNMDOWNER_HPP_
+#define _SRC_INTERFACES_DNMDOWNER_HPP_
+
+#include "internal/dnmd_platform.hpp"
+#include "tearoffbase.hpp"
+#include "controllingiunknown.hpp"
+
+#include <external/cor.h>
+#include <external/corhdr.h>
+
+#include <cstdint>
+#include <atomic>
+
+
+// We use a reference wrapper around the handle to allow the handle to be swapped out.
+// We plan to use swapping to implement table sorting as DNMD itself does not support
+// sorting tables or remapping tokens.
+struct mdhandle_view
+{
+private:
+    mdhandle_ptr& ptr;
```

Comment:
Instead of this, I think it might be better to store a `IDNMDOwner` instance as a field. Then we can easily avoid a dangling reference.